modules/gltf_blend/editor_scene_importer_blend.cpp

```diff
@@ -0,0 +1,113 @@
+
+#include "core/core_bind.h"
+#include "core/object/ref_counted.h"
+#include "modules/gltf/editor_scene_importer_gltf.h"
+#include "scene/3d/node_3d.h"
+#include "scene/animation/animation_player.h"
+#include "scene/resources/animation.h"
+
+uint32_t EditorSceneFormatImporterBlend::get_import_flags() const {
+	return ImportFlags::IMPORT_SCENE;
+}
+
+void EditorSceneFormatImporterBlend::get_extensions(List<String> *r_extensions) const {
+	r_extensions->push_back("blend");
+}
+
+Node *EditorSceneFormatImporterBlend::import_scene(const String &p_path,
+		uint32_t p_flags, int p_bake_fps,
+		List<String> *r_missing_deps,
+		Error *r_err) {
+	String addon_path = EditorSettings::get_singleton()->get_setting("filesystem/blend/blender_path");
+	String scene_to_blend_setting = "import bpy, os, sys;"
+									"bpy.ops.wm.open_mainfile(filepath='GODOT_SOURCE');"
+									"(bpy.ops.export_scene.gltf("
+									"filepath='GODOT_SINK',"
+									"export_format='GLTF_SEPARATE',"
+									"export_keep_originals=True,"
+									"export_texture_dir='GODOT_TEXTURE_DIR',"
+									"export_colors=True,"
+									"export_all_influences=False,"
+									"export_extras=True,"
+									"export_cameras=True,"
+									"export_lights=True)) if bpy.app.version >= (3, 0, 0) else "
+									"(bpy.ops.export_scene.gltf("
+									"filepath='GODOT_SINK',"
+									"export_format='GLTF_SEPARATE',"
+									"export_texture_dir='GODOT_TEXTURE_DIR',"
+									"export_colors=True,"
+									"export_all_influences=False,"
+									"export_extras=True,"
+									"export_cameras=True,"
+									"export_lights=True));";
+	String script = scene_to_blend_setting;
+	String source = ProjectSettings::get_singleton()->globalize_path(p_path);
+	String source_global = ProjectSettings::get_singleton()->globalize_path(source);
+	source_global = source_global.c_escape();
+	String texture_global = "res://.godot/imported/" + p_path.get_file().get_basename() + "-" + p_path.md5_text() + "_textures";
+	texture_global = ProjectSettings::get_singleton()->globalize_path(texture_global);
+	texture_global = texture_global.c_escape();
+	script = script.replace("GODOT_TEXTURE_DIR", texture_global);
+	script = script.replace("GODOT_SOURCE", source_global);
+	String sink = "res://.godot/imported/" + p_path.get_file().get_basename() + "-" + p_path.md5_text() + ".gltf";
+	String sink_global = ProjectSettings::get_singleton()->globalize_path(sink);
+	sink_global = sink_global.c_escape();
+	script = script.replace("GODOT_SINK", sink_global);
+	Vector<String> args;
+	args.push_back("--background");
+	args.push_back("--python-expr");
+	args.push_back(script);
+	print_line(script);
+	Array standard_out;
+	int32_t ret = core_bind::OS::get_singleton()->execute(addon_path, args, standard_out, true);
+	for (int32_t line_i = 0; line_i < standard_out.size(); line_i++) {
+		print_line(standard_out[line_i]);
+	}
+	if (ret != OK) {
+		print_error("Blender returned " + itos(ret));
```

Comment:
Do we need to set `r_err` here? 

When testing the plugin, I was using it with a CI build system, which I had forgotten to install blender on. The export printed errors, but since it returned `0` the build system reported a success, and I never knew to look for the errors.

There doesn't seem to be a way for a `gdscript` plugin to report an error, but I'm wondering if `r_err` will do that here?

Reply:
I'll set r_err.